unit_tests/CMakeLists.txt

```diff
@@ -0,0 +1,63 @@
+#
+# Copyright (C) 2023 The Falco Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+# the License. You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+#
+
+message(STATUS "Falco unit tests build enabled")
+
+include(FetchContent)
+
+FetchContent_Declare(
+  googletest
+  GIT_REPOSITORY https://github.com/google/googletest.git
+  GIT_TAG        release-1.12.1
+)
+
+FetchContent_MakeAvailable(googletest)
+
+file(GLOB_RECURSE ENGINE_TESTS ${CMAKE_CURRENT_SOURCE_DIR}/engine/*.cpp)
+file(GLOB_RECURSE FALCO_TESTS ${CMAKE_CURRENT_SOURCE_DIR}/falco/*.cpp)
+
+set(FALCO_UNIT_TESTS_SOURCES
+  "${ENGINE_TESTS}"
+  "${FALCO_TESTS}"
+)
+
+set(FALCO_UNIT_TESTS_INCLUDES
+    PRIVATE
+    ${CMAKE_SOURCE_DIR}/userspace
+    ${CMAKE_BINARY_DIR}/userspace/falco # we need it to include indirectly `config_falco.h` file
+    ${CMAKE_SOURCE_DIR}/userspace/engine # we need it to include indirectly `falco_common.h` file
+)
+
+set(FALCO_UNIT_TESTS_DEPENDENCIES
+  gtest
+  gtest_main
+  falco_engine
```

Comment:
Reminder for ourselves: we don't need to address this in this PR, however we will not be able to test sources under the `userspace/falco` directory if we don't create a cmake library target for it. The reason why we're able to have `test_configuration.cpp` is that the falco config helpers are header-only.

My proposal for a future PR would be to have a new library target alongside `falco_engine`, and then link `falco.cpp` with both to have the final `falco` executable target. Would it make sense?